styx-scheduler-service/src/main/java/com/spotify/styx/ServiceAccountKeyManager.java

```diff
@@ -68,8 +68,8 @@ public boolean keyExists(String keyName) throws IOException {
           .execute();
       return true;
     } catch (GoogleJsonResponseException e) {
-      // TODO: handle 403 correctly once google fixes their API
-      if (e.getStatusCode() == 403 || e.getStatusCode() == 404) {
+      // TODO: handle 403 and 400 correctly once google fixes their API
+      if (e.getStatusCode() == 403 || e.getStatusCode() == 400 || e.getStatusCode() == 404) {
```

Comment:
It is more accurate to check status to be FAILED_PRECONDITION in case of 400 because 400 could mean different things.